File: linear_algebra/Python/UNK_BT9.py
A=[] 

n=int(input("nhap n: "))
print("nhap ma tran A:")
for i in range(n):
    row=[]
    for j in range(n):
        row.append(int(input(f"A[{i}][{j}] = ")))
    A.append(row)

print('Ma tran A:')
for i in range(n):
    for j in range(n):
        print(A[i][j], end=' ')
    print()

duongcheo = True
tamGiactren = True
tamGiacduoi = True
doixung = True

for i in range(n):
    for j in range(n):
        if A[i][j] !=0 and i!=j:
            duongcheo = False
        if A[i][j] !=0 and i>j:
            tamGiactren = False
        if A[i][j] !=0 and i<j:
            tamGiacduoi = False
        if A[i][j] != A[j][i]:
            doixung = False
print(f"Ma tran duong cheo {'true' if duongcheo else 'false'}")
print(f"Ma tran tam giac tren {'true' if tamGiactren else 'false'}")
print(f"Ma tran tam giac duoi {'true' if tamGiacduoi else 'false'}")
print(f"Ma tran doi xung {'true' if doixung else 'false'}")  

File: linear_algebra/C++/UNK_BT9.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    int A[100][100];
    cout << "Nhap so nguyen duong n: ";
    cin >> n;
    if (n <= 0) {
        cout << "n phai la so nguyen duong";
        return 0;
    }
    cout << "Nhap ma tran A:" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << "A[" << i << "][" << j << "] = ";
            cin >> A[i][j];
        }
        cout << endl;
    }
    cout << "Ma tran" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << A[i][j] << " ";
        }
        cout << endl;
    }
    bool duongcheo = true;
    bool tamGiacTren = true;
    bool tamGiacDuoi = true;
    bool doixung = true;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (A[i][j] != 0 && i != j) {
                duongcheo = false;
            }
            if (A[i][j] != 0 && i < j) {
                tamGiacDuoi = false;
            }
            if (A[i][j] != 0 && i > j) {
                tamGiacTren = false;
            }
            if (A[i][j] != A[j][i]) {
                doixung = false;
            }
        }
    }
    cout << "Ma tran duong cheo: " << (duongcheo ? "true" : "false") << endl;
    cout << "Ma tran tam giac tren: " << (tamGiacTren ? "true" : "false") << endl;
    cout << "Ma tran tam giac duoi: " << (tamGiacDuoi ? "true" : "false") << endl;
    cout << "Ma tran doi xung: " << (doixung ? "true" : "false") << endl;

    return 0;


}